.github/workflows/docker-build-push.yml

```diff
@@ -37,47 +37,32 @@ jobs:
           git checkout $pages_branch
           git merge -s recursive --strategy-option=theirs ${{ github.event.after }}
 
-      # Test if Dockerfile has changed
-      # sets steps.check_docker.outputs.changed to 1 if the Dockerfile has changed, 0 otherwise
-      - name: Check Dockerfile for changes
-        id: check_docker
-        env:
-          BEFORE: ${{ github.event.before }}
-        run: |
-          git diff-index --name-only $BEFORE > changes.txt
-          if grep "docker/Dockerfile" changes.txt ; then
-            echo "Dockerfile changed"
-            echo "::set-output name=changed::1"
-          else
-            echo "No change to Dockerfile"
-            echo "::set-output name=changed::0"
-          fi
-          rm changes.txt
-
       # Login to Dockerhub
       - name: Login to DockerHub
-        if: steps.check_docker.outputs.changed == 1
         uses: docker/login-action@v1
         with:
           username: ${{ secrets.DOCKER_ID }}
           password: ${{ secrets.DOCKER_PASSWORD }}
       # set up Docker build
       - name: Set up Docker Buildx
-        if: steps.check_docker.outputs.changed == 1
         uses: docker/setup-buildx-action@v1
-      # Build docker image (We are not using caching here to force a clean build)
-      - name: Build and Push Docker image
-        if: steps.check_docker.outputs.changed == 1
+      # Build the Docker image
+      - name: Build and Load Docker image
         uses: docker/build-push-action@v2
         with:
-          push: true
+          push: false
+          load: true
           context: docker
           file: docker/Dockerfile
           tags: ccdl/refinebio-examples:latest
+      # push the Docker image if this is master
+      - name: Push Docker image
+        if: github.ref == 'refs/heads/master'
+        run: docker push ccdl/refinebio-examples:latest
```

Comment:
Because users (at this point) are never instructed to use the docker image, and it's mainly for development purposes, isn't it more important that the docker image is up to date at the "staging" stage?

Reply:
I'm not sure what you mean here... This process will also apply to changes to the Docker image, so we need to use the Dockerfile that is part of staging, which may be ahead of the "released" version. I would expect the image on Dockerhub to correspond to the current public pages. If we want to have a `staging` tagged file as well, that would be easy to implement. 

But I think this gets a bit into the process for development: I can imagine where two people working independently would end up on different image versions, regardless of what is listed as 'latest'. We've certainly seen this happen, and I think restricting the `latest` version to be parallel to `master` is the most straightforward to explain.

Reply:
What is someone returning to the project to develop, while others have been actively developing (e.g., people are still merging to staging), supposed to pull from Dockerhub? I think that is the nature of @cansavvy's question.

Reply:
> I think restricting the latest version to be parallel to master is the most straightforward to explain.

Why is this more straightforward than having the docker image updated for `staging` and `master`? 

For development purposes it makes sense to me to have the docker image updates available to pull as soon as they are approved and being used for rendering. 

But we can also make a `staging` tag in preparation for a future time in which we may actually do instruct users to use the docker image.

Reply:
Okay, I think that makes sense. In that case, maybe we want to switch things around: keep `latest` as the tag that corresponds to the staging branch, and make a `release` tag or similar that corresponds to `master`.

Reply:
That seems fine to me.